airbyte-webapp-e2e-tests/cypress/integration/connection.spec.ts

```diff
@@ -201,6 +208,54 @@ describe("Connection main actions", () => {
     deleteDestination(destName);
   });
 
+  it("Create a connection, update data in source, show diff modal, reset streams", () => {
+    cy.intercept("/api/v1/web_backend/connections/update").as("updateConnection");
+
+    populateDBSource();
+    const sourceName = appendRandomString(
+      "Test refresh source schema with changed data - connection Postgres source cypress"
+    );
+    const destName = appendRandomString(
+      "Test refresh source schema with changed data - connection Local JSON destination cypress"
+    );
+
+    createTestConnection(sourceName, destName);
+    cy.get("div").contains(sourceName).should("exist");
+    cy.get("div").contains(destName).should("exist");
+
+    makeChangesInDBSource();
+    openSourceDestinationFromGrid(sourceName);
+    goToReplicationTab();
+    refreshSourceSchemaBtnClick();
+
+    cy.get("[data-testid='catalog-diff-modal']").should("exist");
+
+    cy.get("table[aria-label='removed streams table']").should("contain", "users");
+
+    cy.get("table[aria-label='new streams table']").should("contain", "cars");
+
+    cy.get("button[aria-label='toggle accordion']").click();
```

Comment:
can we move all elements to the pages level and create methods for working with these elements? so everyone can reuse it and we won't duplicate code

Reply:
Duly noted! Will fix that! thanks)